controllers/metallb_controller.go

```diff
@@ -81,9 +83,31 @@ func (r *MetallbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		// Error reading the object - requeue the request.
 		return ctrl.Result{}, err
 	}
+	err = status.UpdateStatus(context.TODO(), r.Client, instance, status.ConditionProgressing, "", "")
+	if err != nil {
+		logger.Info("Failed to update metallb status", "Desired status", status.ConditionProgressing)
+	}
 
 	err = r.syncMetalLBResources(instance)
 	if err != nil {
+		err2 := status.UpdateStatus(context.TODO(), r.Client, instance, status.ConditionDegraded, "FailedToSyncMetalLBResources", err.Error())
+		if err2 != nil {
+			logger.Error(err2, "Failed to update metallb status", "Desired status", status.ConditionDegraded)
+		}
+		return ctrl.Result{}, err
+	}
+	isReady, err := status.IsMetallbAvailable(context.TODO(), r.Client)
+	if err != nil {
+		logger.Error(err, "Failed to check metallb deployment status")
+		return ctrl.Result{}, err
+	}
+	if !isReady {
+		return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
```

Comment:
Can you drop an info log here? Or even better, having IsMetallbAvailable return a typed error, asserting that instead of isReady and dumping it's value.

Reply:
done